databuilder/query_model/population_validation.py

```diff
@@ -211,6 +211,8 @@ def in_(lhs, rhs):
 register_op(Function.Add)(operator.add)
 register_op(Function.Subtract)(operator.sub)
 register_op(Function.Multiply)(operator.mul)
+register_op(Function.TrueDivide)(operator.truediv)
+register_op(Function.FloorDivide)(operator.floordiv)
```

Comment:
I'm not sure what operator to register here because `operator.floordiv` doesn't exist and we need to define our custom one?

Reply:
or does that happen in `databuilder/query_language.py`? I don't think so, I'm confused


```
    def __floordiv__(self, other):
        other = self._cast(other)
        return _apply(qm.Function.FloorDivide, self, other)
```

Reply:
It might be helpful to look at this commit here which introduced the `date_utils` module: https://github.com/opensafely-core/databuilder/commit/e44c327244a22bd63e5ae665b15729ae8b1269cf

Structurally, what we're trying to do here is the same. So I think probably the thing to do is to add a new `maths_utils` module which defines our own `floordiv` function. That could be then be imported and registered in `population_validation.py`:
```py
register_op(Function.FloorDivide)(maths_utils.floordiv)
```

And in `in_memory.py`:
```py
def visit_FloorDivide(self, node):
    return self.visit_binary_op_with_null(node, maths_utils.floordiv)
```